sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/address/ParameterableImpl.java

```diff
@@ -151,16 +157,117 @@ public void setParameter(String name, String value) {
 		if(name.equalsIgnoreCase("branch") && isModifiable == ModifiableRule.Via) {
 			throw new IllegalStateException("it is forbidden to set the branch parameter on the Via Header");
 		}
-		//Fix from abondar for Issue 494 and angelo.marletta for Issue 502      
+		//Fix from abondar for Issue 494 and angelo.marletta for Issue 502	  
 		this.parameters.put(name.toLowerCase(), value);
 		if(header != null) {
 			try {
-				header.setParameter(name, "".equals(value) ? null : value);
+				if (isQuotableParameter(name, value)){
+					header.setQuotedParameter(name, value);
+				}
+				else{
+					header.setParameter(name, "".equals(value) ? null : value);
+				}
 			} catch (ParseException e) {
 				throw new IllegalArgumentException("Problem setting parameter",e);
 			}
 		}
 	}
+	
+	private static final String QUOTABLE_PARAMETER_NAME ="org.restcomm.servlets.sip.QUOTABLE_PARAMETER_NAME";
+	/**
+	 * Verify that the parameter is in the known list of parameters that its value need to be quoted
+	 * 
+	 * @return 
+	 * true: if the Parameter name is present at servletContext and its value can be quoted, else return false
+	 * otherwise
+	 */
+	protected boolean isQuotableParameter(String name, String value){
+		if (sipSession != null && sipSession.getServletContext() != null){
+			// get list of parameter names which are splited by comma.
+			String quotableParamNames = sipSession.getServletContext().getInitParameter(QUOTABLE_PARAMETER_NAME);
+			if (quotableParamNames != null){
+				String[] paramNames = quotableParamNames.split(",");

```

Comment:
JEE specification doesnt allow the user to change the descriptor while the service is running. They may undeploy the app, change the descriptor,and deploy it again. That may survive the same JVM instance,so "static" could be risky here. We may save the original string and compare with current one, if different reparse...